resources/aws_ec2.go

```diff
@@ -820,9 +830,37 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 		}
 		log.Printf("%s: instance running", obj)
 	}
+	// get the instance's public ip
+	ip, err := getPublicIP(obj.client, instanceID)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error getting public ip")
+	}
+	// store it in the struct for other resources to receive
```

Comment:
Don't be fooled at how small the amount of code is for send/recv, as this is incorrect! How come? Well you might imagine that mgmt starts up and checks the state, and the ec2 instance is already in the correct state and we return from CheckApply with (true, nil) much earlier in the function! In this situation, the IPAddress was never set!

So you need to make sure that it is always set anytime before the function returns, whenever it could return and we'd have wanted to update it. (If the machine's IP disappears, we probably want to set it to empty!)